service/gateway/request_util.go

```diff
@@ -75,6 +77,8 @@ func newRequestContext(r *http.Request) *requestContext {
 		bucketName: vars["bucket"],
 		objectName: vars["object"],
 		accountID:  vars["account_id"],
+		maxKeys:    r.URL.Query().Get("max_keys"),
+		startAfter: r.URL.Query().Get("start_after"),
```

Comment:
The parts that are not common can be parsed in the corresponding handler.

Reply:
fixed